all/experiments/writer.py

```diff
@@ -5,11 +5,18 @@
 from datetime import datetime
 from tensorboardX import SummaryWriter
 from all.logging import Writer
+from comet_ml import Experiment
+
+COMET_AVAILABLE = True
+try:
+    from comet_ml import Experiment
+except ImportError:
+    COMET_AVAILABLE = False
```

Comment:
It would probably be better to include this block inside the constructor for `CometWriter`.